app/controllers/subscriptions_controller.rb

```diff
@@ -17,7 +17,7 @@ def create
       SendEmailWorker.perform_async_in_queue(email.id, queue: :send_email_transactional)
     end
 
-    render json: { id: subscription.id }
+    render json: { id: subscription.id, subscription: subscription }
```

Comment:
Are we just going to be using the frequency of the subscription? I wonder whether we're being too leaky by rendering the entire subscription object rather than just picking the bits that we want? It might make it hard for us to refactor at a later date if we need to.

Reply:
The plan is actually to use the subscriber ID, which is embedded in this object (due to the way `as_json` recurses through ActiveRecord associations). While it is more data than we need, it's:

- Consistent with the way we respond to [an update](https://github.com/alphagov/email-alert-api/blob/master/app/controllers/subscriptions_controller.rb#L78).

- Being selective here in the past is making it harder to change, since we didn't use a nested hash.

A compromise might be something like (after we remove the redundant, top-level ID key):

```
render json: { subscription: { id: <id>, subscriber: { id: <subscriber_id> } } }
```

Since we're already returning the whole thing on the "update" action, though, I reckon this approach is more surprising than just being consistent with the code below it.

What do you think?

Reply:
Cool, thanks for the explanation. I might feel different if this was a public API, but it isn't - we control all consumers of it. Being consistent is probably more important then, I reckon, and it's a good point about our previous choices making it annoying.

Happy to go with what we've got 👍

Reply:
Oh, yes that's a good point. I guess you've got quite a different perspective on it from e.g. your work on search (API). I'll bear that in mind when I'm working on APIs in the future - thanks!